pruner/pruner.go

```diff
@@ -0,0 +1,41 @@
+package pruner
+
+import (
+	"context"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// AvailabilityWindow defines an interface for pruning historical data.
+type AvailabilityWindow interface {
+	// IsWithinAvailabilityWindow returns true if the given header is within the
+	// "availability" (sampling) window of the network.
+	IsWithinAvailabilityWindow(eh *header.ExtendedHeader, window time.Duration) bool
+}
+
+// Pruner handles the pruning routine for the node using the
+// prune Factory.
+type Pruner struct {
+	factory Factory
+}
+
+func NewPruner(factory Factory) *Pruner {
+	return &Pruner{
+		factory: factory,
+	}
+}
+
+func (p *Pruner) Start(context.Context) error {
```

Comment:
would be good to expose some method on factory / or a config to enable/disable the pruning routine in case it's not needed (for archival, for example)

Reply:
Like it immediately quits out? I don't think it's necessary. I'd rather have some sort of way to short-circuit the start + stop if there's no "active prune factory" found. We also usually like to have start and stop conform to the lifecycle iface